src/js/drawer.js

```diff
@@ -68,19 +68,28 @@ function handleCloseEvents (scope, callback) {
 	return { addEvents, removeEvents, handleMouseleave };
 }
 
-function addDrawerToggles (drawerEl) {
+function addDrawerToggles (drawerEl, allFocusable) {
 	const controls = Array.from(document.body.querySelectorAll(`[aria-controls="${drawerEl.id}"]`));
 
 	let handleClose;
 	let openingControl;
 
+	const handleTab = (e) => {
+		if (e.keyCode === 9 && e.target === allFocusable[allFocusable.length - 1]) {
+			// Keep focus within the drawer when tabbing for a11y reasons.
+			drawerEl.focus();
+			e.preventDefault();
+		}
+	};
+
 	function toggleCallback (state, e) {
 		if (state === 'close') {
 			toggleTabbing(drawerEl, false);
 
 			handleClose.removeEvents();
 
 			openingControl.focus();
+			drawerEl.removeEventListener('keydown', handleTab);
```

Comment:
I'm not super familiar with `o-header`, but it looks like removing this event listener would belong in `handleClose.removeEvents();`?